src/Features/LanguageServer/Protocol/Workspaces/LspMiscellaneousFilesWorkspace.cs

```diff
@@ -100,22 +100,5 @@ public ValueTask UpdateTextIfPresentAsync(DocumentId documentId, SourceText sour
             this.OnDocumentTextChanged(documentId, sourceText, PreservationMode.PreserveIdentity, requireDocumentPresent: false);
             return ValueTaskFactory.CompletedTask;
         }
-
-        private static LanguageInformation? GetLanguageInformation(string documentPath, string languageId)
-        {
-            if (s_extensionToLanguageInformation.TryGetValue(Path.GetExtension(documentPath), out var languageInformation))
-            {
-                return languageInformation;
-            }
-
-            // It is totally possible to not find language based on the file path (e.g. a newly created file that hasn't been saved to disk).
-            // In that case, we use the languageId that the client gave us.
-            return languageId switch
-            {
-                "csharp" => s_csharpLanguageInformation,
-                "vb" => s_vbLanguageInformation,
-                _ => null,
-            };
```

Comment:
well now i'm unhappy about where we were before as well :'(

Reply:
Yeah, I pretty much just moved this logic